src/Cryptol/Symbolic/SBV.hs

```diff
@@ -110,11 +111,39 @@ proverConfigs =
 setTimeoutSecs ::
   Int {- ^ seconds -} ->
   SBV.SMTConfig -> SBV.SMTConfig
-setTimeoutSecs s cfg = cfg
-  { SBV.solverSetOptions =
-    SBV.OptionKeyword ":timeout" [show (toInteger s * 1000)] :
-    filter (not . isTimeout) (SBV.solverSetOptions cfg) }
+setTimeoutSecs s cfg = case SBV.name (SBV.solver cfg) of
+  SBV.Yices -> cfg'
+  -- TODO: although "--timeout" correctly sets the timeout for Yices,
+  -- SBV crashes due to an unexpected response from Yices after
+  -- a timeout is hit.
+  -- As a workaround, we ignore the timeout setting for Yices
+  -- and instead rely on addDeadmanTimer to kill Yices when the
+  -- timeout is hit.
+  -- see:
+  --   https://github.com/LeventErkok/sbv/issues/735
+  --   https://github.com/SRI-CSL/yices2/issues/547
+    {- { SBV.extraArgs =
+      ["--timeout", show (toInteger s)] ++
+      SBV.extraArgs cfg' } -}
+  -- NOTE: in sbv 11.1.5 these special cases are handled
+  -- by a new 'SetTimeOut' constructor for 'SMTOption'
```

Comment:
Is this comment meant to apply to the Yices timeout above, or the CVC{4,5} timeouts below? If it's the latter, it would be worth adding a newline between this comment and the Yices-specific comments above to make this more obvious.

Also, it would be worth leaving a comment saying that once Cryptol requires `sbv-11.1.5` as the minimum, then we can drop the CVC{4,5}-related special cases below.